File: second_innings/lib/services/api_response.dart
class ApiResponse<T> {
  final int statusCode;
  final T? data;
  final String? message;
  final String? error;
  final bool isSuccess;

  ApiResponse({required this.statusCode, this.data, this.message, this.error})
    : isSuccess = statusCode >= 200 && statusCode < 300;

  factory ApiResponse.success({
    required int statusCode,
    T? data,
    String? message,
  }) {
    return ApiResponse<T>(statusCode: statusCode, data: data, message: message);
  }

  factory ApiResponse.error({
    required int statusCode,
    String? error,
    String? message,
  }) {
    return ApiResponse<T>(
      statusCode: statusCode,
      error: error ?? message,
      message: message,
    );
  }

  @override
  String toString() {
    return 'ApiResponse(statusCode: $statusCode, isSuccess: $isSuccess, data: $data, error: $error)';
  }
}

File: admin_web/src/utils/apiResponse.js
export class ApiResponse {
  constructor({ statusCode, data = null, message = null, error = null }) {
    this.statusCode = statusCode
    this.data = data
    this.message = message
    this.error = error
    this.isSuccess = statusCode >= 200 && statusCode < 300
  }

  static success({ statusCode, data = null, message = null }) {
    return new ApiResponse({ statusCode, data, message })
  }

  static error({ statusCode, error = null, message = null }) {
    return new ApiResponse({
      statusCode,
      error: error || message,
      message,
    })
  }

  toString() {
    return `ApiResponse(statusCode: ${this.statusCode}, isSuccess: ${this.isSuccess}, data: ${this.data}, error: ${this.error})`
  }
}
